src/actions/generate-fixtures.ts

```diff
@@ -100,42 +80,90 @@ const generateIdentity = async (mnemonic: string) => {
         //
     }
 
-    // Write identity to JSON file
-    await writeIdentityToJson(identity);
+    await writeFixtureToFile("identity.json", identity);
     console.log("\nIdentity data written to data/identity.json");
 };
 
-const generateTransfers = async (mnemonic: string) => {
-    const app = getApplication();
-
-    const signed = await app
-        .resolve(EvmCallBuilder)
-        .gasPrice(5000000000)
-        .gasLimit(21000)
-        .nonce("1")
-        .recipientAddress("0x6F0182a0cc707b055322CcF6d4CB6a5Aff1aEb22")
-        .value("100000000")
-        .payload("")
-        .sign(mnemonic);
-
-    const builtTransaction = await signed.build();
-
-    const transferData = {
+const getTransactionData = (builtTransaction: object) => {
+    return {
         data: Object.fromEntries(
-            Object.entries(builtTransaction.data).map(([key, value]) => [
+            Object.entries(builtTransaction['data']).map(([key, value]) => [
                 key,
                 // Convert BigNumber objects to strings, keep other values as is
                 value && typeof value === "object" && "toString" in value ? value.toString() : value,
             ]),
         ),
-        serialized: builtTransaction.serialized.toString("hex"),
+        serialized: builtTransaction['serialized'].toString("hex"),
     };
+}
+
+const generateTransaction = async (mnemonic: string, fixtureName: string, config: object) => {
+    const app = getApplication();
 
-    // @ts-ignore
-    await writeTransferToJson(transferData);
-    console.log("Transfer data written to data/transfer.json");
+    const transaction = app
+        .resolve(EvmCallBuilder)
+        .gasPrice(config["gasPrice"] || 5000000000)
+        .gasLimit(config["gasLimit"] || 21000)
+        .nonce(config["nonce"] || "1")
+        .value(config["value"] || "0")
+        .payload("");
+
+    if (config["recipientAddress"]) {
+        transaction.recipientAddress(config["recipientAddress"]);
+    }
+
+    const signed = await transaction.sign(mnemonic);
+
+    const builtTransaction = await signed.build();
+
+    await writeFixtureToFile(`${fixtureName}.json`, getTransactionData(builtTransaction));
+    console.log(`Transfer data written to data/${fixtureName}.json`);
 };
 
+const generateContract = async (mnemonic: string, fixtureName: string, config: object) => {
```

Comment:
at first I read this as generating a contract (deployment) transaction. Maybe better to name it `generateTransaction` (when the other is named `generateTransfer`) as it generates various EVM transactions?